app/gui/view/src/project.rs

```diff
@@ -376,23 +398,54 @@ impl View {
         let input_change_delay = frp::io::timer::Timeout::new(network);
         let searcher_open_delay = frp::io::timer::Timeout::new(network);
 
-        if let Some(window_control_buttons) = &*model.window_control_buttons {
-            let initial_size = &window_control_buttons.size.value();
-            model.graph_editor.input.space_for_window_buttons(initial_size);
+        frp::extend! { network
+            init <- source_();
+        }
+
+
+        // === Top Bar ===
+
+        let window_control_buttons_width = if let Some(window_control_buttons) =
+            &*model.window_control_buttons
+        {
             frp::extend! { network
-                graph.space_for_window_buttons <+ window_control_buttons.size;
-                eval_ window_control_buttons.close      (model.on_close_clicked());
+                window_control_buttons_size <- all(init, window_control_buttons.size)._1();
+                window_control_buttons_width <- window_control_buttons_size.map(|size| size.x);
+                eval_ window_control_buttons.close (model.on_close_clicked());
                 eval_ window_control_buttons.fullscreen (model.on_fullscreen_clicked());
             }
+            window_control_buttons_width
+        } else {
+            frp::extend! { network
+                window_control_buttons_width <- init.constant(crate::graph_editor::TOP_BAR_ITEM_MARGIN);
+            }
+            window_control_buttons_width
+        };
```

Comment:
Why is this an anti-pattern? What's the negative consequence of having the FRP network be based on a constant that won't change for the duration of the application?

I can try to change it though. I just kept the conditional creation of FRP nodes that already there in the code.

Reply:
Hmmm, I don't see how to do something like that. So we're looking for a way to react to a change of an FRP node if it exists (`window_control_buttons.size`) or use a constant otherwise.  Is there some FRP primitive that sort of thing?

Is the condition `eval` still fine?
```rust
        if let Some(window_control_buttons) = &*model.window_control_buttons {
            frp::extend! { network
                eval_ window_control_buttons.close (model.on_close_clicked());
                eval_ window_control_buttons.fullscreen (model.on_fullscreen_clicked());
            }
        }
```

Reply:
FRP should not be added / removed dynamically because it makes debugging FRP networks really hard. Static networks are just the way to go. 

Regarding the conditional eval, we do not have such code anywhere in the app, as it is still hard to debug. Instead, you should have an frp input nodes `app.close` and `app.fullscreen` and if the window control buttons are created, they should attach to these inputs like `app.close <+ button.pressed`. This way the logic of connecting to the right API is embedded in the buttons creation logic, not outside of them and you don't have conditional FRP creation.

Reply:
Hmmm, doesn't putting code like `app.close <+ button.pressed` in the button definition tightly couple the button code to how it's used? That seems like an anti-pattern to me? So, just to make sure I understand correctly, do you suggest to pass the FRP network that will use the button to the creation/initialization function of the button?

Reply:
No, I'm not talking about "any button definition". I'm talking about this specific button definition.  If button is conditionally created, then the code responsible for its creation can attach its events to another FRP network events by using the `<+` functionality.

Such a code has also other benefits – allows for showing / hiding window control buttons instead of hardcoding this state forever at the app start.

Reply:
Okay, I see, I can move the `eval`s to the place where the buttons get created.

Do you have a solution to this I mentioned earlier? I don't see it.
> Hmmm, I don't see how to do something like that. So we're looking for a way to react to a change of an FRP node if it exists (`window_control_buttons.size`) or use a constant otherwise. Is there some FRP primitive that sort of thing?